pkg/collector/corechecks/system/memory_windows.go

```diff
@@ -25,17 +28,72 @@ var runtimeOS = runtime.GOOS
 // MemoryCheck doesn't need additional fields
 type MemoryCheck struct {
 	core.CheckBase
+	cacheBytes     *pdhutil.PdhCounterSet
+	committedBytes *pdhutil.PdhCounterSet
+	pagedBytes     *pdhutil.PdhCounterSet
+	nonpagedBytes  *pdhutil.PdhCounterSet
 }
 
 const mbSize float64 = 1024 * 1024
 
+// Configure handles initial configuration/initialization of the check
+func (c *MemoryCheck) Configure(data integration.Data, initConfig integration.Data) (err error) {
+	c.cacheBytes, err = pdhutil.GetCounterSet("Memory", "Cache Bytes", "", nil)
+	if err == nil {
+		c.committedBytes, err = pdhutil.GetCounterSet("Memory", "Committed Bytes", "", nil)
+		if err == nil {
+			c.pagedBytes, err = pdhutil.GetCounterSet("Memory", "Pool Paged Bytes", "", nil)
+			if err == nil {
+				c.nonpagedBytes, err = pdhutil.GetCounterSet("Memory", "Pool Nonpaged Bytes", "", nil)
+			}
+		}
+	}
+	return err
+}
+
 // Run executes the check
 func (c *MemoryCheck) Run() error {
 	sender, err := aggregator.GetSender(c.ID())
 	if err != nil {
 		return err
 	}
 
+	var val float64
+	if c.cacheBytes != nil {
+		val, err = c.cacheBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.cached", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.cached")
+		}
+	}
+
+	if c.committedBytes != nil {
+		val, err = c.committedBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.committed", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.committed")
+		}
+	}
+
+	if c.pagedBytes != nil {
+		val, err = c.pagedBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.paged", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.paged")
+		}
+	}
+
+	if c.nonpagedBytes != nil {
+		val, err = c.nonpagedBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.nonpaged", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.nonpaged")
```

Comment:
Intuitively I'd say it'd be useful to add the error to the logged message, in order to have more info on what failed exactly

(applies to all the logging calls added in this function)

Reply:
also, the logic at the end of the function:

```
if errVirt != nil && errSwap != nil {
    return fmt.Errorf("failed to gather any memory information")
}
```

needs to be updated